src/Storages/StoragePostgreSQL.cpp

```diff
@@ -456,15 +458,20 @@ SinkToStoragePtr StoragePostgreSQL::write(
     return std::make_shared<PostgreSQLSink>(metadata_snapshot, pool->get(), remote_table_name, remote_table_schema, on_conflict);
 }
 
-StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, bool require_table)
+StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, PostgreSQLSettings& storage_settings, bool require_table)
 {
     StoragePostgreSQL::Configuration configuration;
+
+    ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> optional_arguments = {"replace_query", "on_duplicate_clause", "addresses_expr", "host", "hostname", "port"};
+    auto postgresql_settings = storage_settings.all();
+    for (const auto & setting : postgresql_settings)
+        optional_arguments.insert(setting.getName());
```

Comment:
Allowing postgresql engine settings in named collection here is pointless because even though you add them here to `optional_arguments_list` (which means that these settings will pass "arguments validation"), but their values are never used/parsed anywhere. Which means the user will be able to specify them but they will not work. 

So let's either remove this code or make a proper support. It can be done similar like it is done in mysql https://github.com/ClickHouse/ClickHouse/blob/621d8522892e689d4049ae8ab131e30f97137859/src/Storages/StorageMySQL.cpp#L294-L299.